skbio/stats/composition.py

```diff
@@ -807,6 +807,319 @@ def centralize(mat):
+    """
+
+    # Log transformation
+    x = np.log(x)
+    y = np.log(y)
+    
+    #Variance log ratio
+    return np.var(x - y, ddof=ddof)
+
+
+@experimental(as_of="0.5.7")
+def _robust_vlr(x:np.array, y:np.array, ddof:int):
+    r"""
+    Calculates variance log ratio while masking zeros
+    
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+        
+    ddof: int
+        degrees of freedom
+        
+    Returns
+    -------
+    float
+         variance log ratio value
+         
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Mask zeros
+    x = np.ma.masked_array(x, mask=x == 0)
+    y = np.ma.masked_array(y, mask=y == 0)
+    
+    # Log transformation
+    x = np.ma.log(x)
+    y = np.ma.log(y)
+    
+    #Variance log ratio
+    return np.ma.var(x - y, ddof=ddof)
+
+@experimental(as_of="0.5.7")
+def vlr(x:np.array, y:np.array, ddof:int=1, robust:bool=False):
+    r"""
+    Calculates variance log ratio
+    
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+        
+    ddof: int
+        degrees of freedom
+        
+    robust: bool
+        mask zeros at the cost of performance
+       
+    Returns
+    -------
+    float
+         variance log ratio value
+         
+    Examples
+    --------
+    No zeros
+    >>> x = [1,2,3]
+    >>> y = [5,8,13]
+    >>> %timeit vlr(x,y)
+    >>> 0.01277962183258352
+
+    Zeros without robust
+    >>> x = [1,2,3,0]
+    >>> y = [5,8,13,21]
+    >>> vlr(x,y)
+    >>> nan
+
+    Zeros with robust
+    >>> x = [1,2,3,0]
+    >>> y = [5,8,13,21]
+    >>> vlr(x,y, robust=True)
+    >>> 0.01277962183258352
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+    # Convert array_like to numpy array
+    x = np.asarray(x)
+    y = np.asarray(y)
+
```

Comment:
We probably should have some checks to make sure that the inputs are sane (i.e. greater or equal to zero).
It'll be great to have a unittest that also accommodates this as well.

Reply:
Are you thinking something like this? 

```python
x = np.asarray(x)
y = np.asarray(y)

assert np.all(x >= 0)
assert np.all(y >= 0)
```

Reply:
yes, `ValueError` would be preferable since it provides an informative message.